### PR TITLE
Repairing gha release workflow

### PR DESCRIPTION
**Pull request by dylanbannon**
## what
* Repairing the GitHub Actions workflow that triggers on releases.

## why
* Multiple steps in the workflow ("Push Site to S3" and "Push New Index to Algolia") required additional environmental variables to execute properly, since the make targets they called were not originally intended to run in a GitHub Actions step all by themselves. Additionally, "Push New Index to Algolia" needed changes made to its make target to run successfully in GitHub Actions.

### CODE REVIEW

File: .github/workflows/hugo.yaml [hunk 1]
```diff
@@ -51,30 +51,36 @@ jobs:
...
         S3_BUCKET_NAME: docs.prod.cloudposse.org
         AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
         AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        HUGO_URL: ${{ env.HUGO_URL }}release/${{ steps.version.outputs.tags }}
+        SEMVERSION_TAG: ${{ steps.version.outputs.tags }}
       run: make deploy
     - name: "Push New Index to Algolia"
       if: github.event_name == 'release' && github.event.action == 'published'
       env:
+        HUGO_PUBLISH_DIR: public/release/${{ steps.version.outputs.tags }}
```
Comment (osterman) [inline .github/workflows/hugo.yaml:83]: I think this is the wrong index file. We don't want search pointing to the versioned release. We want it  pointing to what's deployed in /

/release/x.y.z is for historical archives

File: Makefile [hunk 2]
```diff
@@ -29,6 +29,8 @@ export DOCKER_BUILD_FLAGS =
 ifeq ($(wildcard /.dockerenv),)
 export DOCKER_RUN_FLAGS ?= -it --rm
 export DOCKER_RUN := docker run $(DOCKER_RUN_FLAGS) -v $(CURDIR):/src -p $(HUGO_PORT):$(HUGO_PORT) -e YARN_BUILD_DISABLED -e GITHUB_BASIC_AUTH $(DOCKER_IMAGE_NAME)
+export DOCKER_ALGOLIA_ENVS ?= -e ALGOLIA_INDEX_NAME -e ALGOLIA_APP_ID -e ALGOLIA_ADMIN_KEY -e ALGOLIA_INDEX_FILE
```
Comment (osterman) [inline Makefile:32]: If we had to do this for algolia, we probably need it for cloudflare

Comment (dylanbannon) [reply Makefile:32]: I assumed so, too, but the cloudflare target apparently ran with only a few warnings and no errors.

Comment (dylanbannon) [reply Makefile:32]: https://github.com/cloudposse/docs/runs/1843433923

Comment (osterman) [reply Makefile:32]: https://github.com/danielpigott/cloudflare-cli/issues/38

Comment (osterman) [reply Makefile:32]: The warning is indicative of not being able to read the envs. Unfortunately, in the world of software engineering, error checking and documentation uncommon practices. ;)

Comment (osterman) [reply Makefile:32]: (we can also upgrade the cloudflare-cli while we're at it)

File: .github/workflows/hugo.yaml (hunk 1 above)
Comment (dylanbannon) [reply .github/workflows/hugo.yaml]: This was resolved by removing that line, which defaults `HUGO_PUBLISH_DIR` to `public`, the non-versioned website build.

File: Makefile [hunk 3]
```diff
@@ -121,12 +117,14 @@ deploy:
 	aws s3 sync --delete --acl public-read --exact-timestamps $(HUGO_PUBLISH_DIR)/release/$(SEMVERSION_TAG) s3://$(S3_BUCKET_NAME)/release/$(SEMVERSION_TAG)
 
 ## Update algolia search index
+reindex: DOCKER_ENVS += -e ALGOLIA_INDEX_NAME -e ALGOLIA_APP_ID -e ALGOLIA_ADMIN_KEY -e ALGOLIA_INDEX_FILE
```
Comment (osterman) [inline Makefile:120]: `DOCKER_RUN_FLAGS` not `DOCKER_ENVS`

File: Makefile [hunk 4]
```diff
@@ -121,12 +117,14 @@ deploy:
...
 reindex:
-	$(ALGOLIA_DOCKER_RUN) atomic-algolia
+	$(DOCKER_RUN) atomic-algolia
 
 ## Invalidate CloudFlare cache (all files)
+invalidate-cache: DOCKER_ENVS += -e CF_API_EMAIL -e CF_API_KEY -e CF_API_DOMAIN
```
Comment (osterman) [inline Makefile:125]: `DOCKER_RUN_FLAGS` not `DOCKER_ENVS`

File: Makefile [hunk 5]
```diff
@@ -28,11 +28,13 @@ export DOCKER_BUILD_FLAGS =
 
 ifeq ($(wildcard /.dockerenv),)
 export DOCKER_RUN_FLAGS ?= -it --rm
-export DOCKER_RUN := docker run $(DOCKER_RUN_FLAGS) -v $(CURDIR):/src -p $(HUGO_PORT):$(HUGO_PORT) -e YARN_BUILD_DISABLED -e GITHUB_BASIC_AUTH $(DOCKER_IMAGE_NAME)
+export DOCKER_RUN ?= docker run $(DOCKER_RUN_FLAGS) -v $(CURDIR):/src -p $(HUGO_PORT):$(HUGO_PORT) -e YARN_BUILD_DISABLED -e GITHUB_BASIC_AUTH $(DOCKER_ENVS) $(DOCKER_IMAGE_NAME)
```
Comment (osterman) [inline Makefile:31]:
```suggestion
export DOCKER_RUN ?= docker run $(DOCKER_RUN_FLAGS) -v $(CURDIR):/src -p $(HUGO_PORT):$(HUGO_PORT) -e YARN_BUILD_DISABLED -e GITHUB_BASIC_AUTH $(DOCKER_ENVS) $(DOCKER_IMAGE_NAME)
```
```suggestion
export DOCKER_RUN ?= docker run $(DOCKER_RUN_FLAGS) -v $(CURDIR):/src -p $(HUGO_PORT):$(HUGO_PORT) -e YARN_BUILD_DISABLED -e GITHUB_BASIC_AUTH $(DOCKER_IMAGE_NAME)
```
reduce variable polution

File: Makefile [hunk 6]
```diff
@@ -115,10 +117,12 @@ deploy:
 	aws s3 sync --delete --acl public-read --exact-timestamps $(HUGO_PUBLISH_DIR)/release/$(SEMVERSION_TAG) s3://$(S3_BUCKET_NAME)/release/$(SEMVERSION_TAG)
 
 ## Update algolia search index
+reindex: DOCKER_ENVS += -e ALGOLIA_INDEX_NAME -e ALGOLIA_APP_ID -e ALGOLIA_ADMIN_KEY -e ALGOLIA_INDEX_FILE
```
Comment (osterman) [inline Makefile:120]:
```suggestion
reindex: DOCKER_RUN_FLAGS += -e ALGOLIA_INDEX_NAME -e ALGOLIA_APP_ID -e ALGOLIA_ADMIN_KEY -e ALGOLIA_INDEX_FILE
```

File: Makefile [hunk 7]
```diff
@@ -115,10 +117,12 @@ deploy:
...
 reindex:
 	$(DOCKER_RUN) atomic-algolia
 
 ## Invalidate CloudFlare cache (all files)
+invalidate-cache: DOCKER_ENVS += -e CF_API_EMAIL -e CF_API_KEY -e CF_API_DOMAIN
```
Comment (osterman) [inline Makefile:125]:
```suggestion
invalidate-cache: DOCKER_RUN_FLAGS += -e CF_API_EMAIL -e CF_API_KEY -e CF_API_DOMAIN
```

**Review by osterman — APPROVED** [review]

**Pull request merged**